app/models/agents/website_agent.rb

```diff
@@ -106,24 +106,41 @@ class WebsiteAgent < Agent
 
       Set `http_success_codes` to an array of status codes (e.g., `[404, 422]`) to treat HTTP response codes beyond 200 as successes.
 
+      The `interpolate` option is the way to format extracted data, and add meta information to each resulted payload.  Its value must be a hash, whose key-value pairs are interpolated after extraction and added to each event payload.  e.g.:
+
+          "interpolate": {
```

Comment:
:( Somehow GitHub ate my comment from last night.

I agree that it is not intuitive at all, sometimes you need to use the `WebsiteAgent`, sometimes the `PostAgent` we have a few features that are exclusive for one or the other Agent.
Moving from a `WebsiteAgent` to a `PostAgent -> WebsiteAgent` combination should only require a change from `url` to `data_from_event`.
If you feel the tradeoff is too annoying for most people we can go with your changes, but I really think we should split up and merge both Agents. Having the scenario in multiple steps can also make it easier to debug the individual Agents, after fetching the data one can verify it matches the expected result and then focus on the extraction (I am almost always doing that manually with the WebsiteAgent, first extract everything then narrow down the selectors I need).